crates/light-client-prover/src/circuit.rs

```diff
@@ -161,8 +167,19 @@ pub fn run_circuit<DaV: DaVerifier, G: ZkvmGuest>(
                             batch_proof_method_ids[idx].1
                         };
 
-                        if G::verify(&journal, &batch_proof_method_id.into()).is_err() {
-                            // if the batch proof is invalid, continue to the next blob
+                        // we unwrap_or to u32::MAX because if hint is empty, then it means all proofs are expected to pass
+                        if current_proof_index
+                            != expected_to_fail_hints.peek().copied().unwrap_or(u32::MAX)
+                        {
+                            // if index is not in the expected to fail hints, then it should pass
+                            G::verify(&journal, &batch_proof_method_id.into())
+                                .expect("Proof hinted to pass failed");
+                        } else {
+                            // if index is in the expected to fail hints, then it should fail
+                            G::verify_expected_to_fail(&proof, &batch_proof_method_id.into())
+                                .expect_err("Proof hinted to fail passed");
+                            expected_to_fail_hints.next();
+                            current_proof_index += 1;
```

Comment:
```suggestion
if expected_to_fail_hints.next_if(|&x| x == current_proof_index).is_none() {
    // if index is not in the expected to fail hints, then it should pass
    G::verify(&journal, &batch_proof_method_id.into())
        .expect("Proof hinted to pass failed");
} else {
    // if index is in the expected to fail hints, then it should fail
    G::verify_expected_to_fail(&proof, &batch_proof_method_id.into())
        .expect_err("Proof hinted to fail passed");
    current_proof_index += 1;
    continue;
}
```
Can use `next_if` so we don't need to unwrap to u32::MAX.

Reply:
https://github.com/chainwayxyz/citrea/pull/1653/commits/14d2b3ea3a2c074d4962d7c665ed92769dad372b